proxy/src/transparency/server.rs

```diff
@@ -91,6 +95,31 @@ where
         let local_addr = connection.local_addr().unwrap_or(self.listen_addr);
         let proxy_ctx = self.proxy_ctx.clone();
 
+        let disable_protocol_detection = orig_dst
+            .map(|addr| {
+                self.disable_protocol_detection_ports.contains(&addr.port())
+            })
+            .unwrap_or(false);
+
+        if disable_protocol_detection {
```

Comment:
This duplicates the code starting at line 169.  Please refactor this so the code duplication is removed. That way we don't need to keep them in sync.

Reply:
I've refactored into a tiny function below.

Reply:
Thanks. I see that didn't work out quite as well as I expected it would but I guess it will still help with future changes.